community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -268,21 +303,14 @@ synchronized void releaseReadLock( Object tx ) throws LockNotFoundException
                         {
                             // found a write lock with all read locks
                             listItr.remove();
-                            if ( !we.element.movedOn )
-                            {
-                                we.waitingThread.interrupt();
-                                // ----
-                                break;
-                            }
+                            we.waitingThread.interrupt();
```

Comment:
Not really, because those do not serve server for the same reason, movedOn before was used to track can we interrupt thread or not, but since now we or cleanup our waiting task from waiting list or mark waiting task as completed (for cases where it was just interrupted but interrupted thread lost the race for sync. object) i do not see why we need to keep it.